src/pages/signing_official_console/SigningOfficialTable2.jsx

```diff
@@ -0,0 +1,367 @@
+import React from 'react';
+import { useState, useEffect, useCallback, useRef } from 'react';
+import { Checkbox } from '@mui/material';
+import { Styles } from '../../libs/theme';
+import { isEmpty } from 'lodash/fp';
+import SimpleTable from '../../components/SimpleTable';
+import PaginationBar from '../../components/PaginationBar';
+import SearchBar from '../../components/SearchBar';
+import {
+  Notifications,
+  recalculateVisibleTable,
+  getSearchFilterFunctions,
+  searchOnFilteredList
+} from '../../libs/utils';
+import { DAA } from '../../libs/ajax/DAA';
+import {User} from '../../libs/ajax/User';
+import { USER_ROLES } from '../../libs/utils';
+import { Button } from '@mui/material';
+
+//Styles specific to this table
+const styles = {
+  baseStyle: {
+    fontFamily: 'Montserrat',
+    fontSize: '1.45rem',
+    fontWeight: 400,
+    color: 'rgb(53, 64, 82)',
+    display: 'flex',
+    padding: '1rem 2%',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  columnStyle: Object.assign({}, Styles.TABLE.HEADER_ROW, {
+    justifyContent: 'space-between',
+    backgroundColor: '#e2e8f4',
+    fontSize: '1.25rem'
+  }),
+  cellWidths: {
+    email: '25%',
+    name: '20%',
+    libraryCard: '25%',
+    role: '20%',
+  },
+};
+
+//column header format for table
+let columnHeaderFormat = {
+  email: {label: 'Email', cellStyle: {width: styles.cellWidths.email}},
+  name: {label: 'Name', cellStyle: {width: styles.cellWidths.name}, sortable: true},
+  libraryCard: {label: 'Library Card', cellStyle: {width: styles.cellWidths.libraryCard}},
+  role: {label: 'Role', cellStyle: {width: styles.cellWidths.role}},
+};
+
+const researcherFilterFunction = getSearchFilterFunctions().signingOfficialResearchers;
+
+const handleClick = async (researcher, specificDac, filteredDaas, checked, refreshResearchers, setResearchers) => {
+  if (!checked) {
+    try {
+      const daaId = filteredDaas.find(daa => daa.dacs.some(dac => dac.dacId === specificDac.dacId))?.daaId;
```

Comment:
This line can be pulled outside of the surrounding if block since it applies to both the checked and unchecked conditions